### PR TITLE
Log gradients

### PR DESCRIPTION
**Pull request by peasant98**
Log gradients from agents.

### CODE REVIEW

File: pfrl/agents/hiro_high_level_goal_conditioned_td3.py [hunk 1]
```diff
@@ -184,6 +184,11 @@ def high_level_update_q_func_with_goal(self, batch):
         self.q_func1_loss_record.append(float(loss1))
         self.q_func2_loss_record.append(float(loss2))
 
+        q1_recent_variance = np.var(list(self.q1_record)[-100:])
+        q2_recent_variance = np.var(list(self.q2_record)[-100:])
```
Comment (watakandai) [inline pfrl/agents/hiro_high_level_goal_conditioned_td3.py]: why 100? Give it a name for this constant.

File: pfrl/agents/goal_conditioned_td3.py [hunk 2]
```diff
@@ -174,7 +188,13 @@ def update_q_func_with_goal(self, batch):
         self.q_func1_loss_record.append(float(loss1))
         self.q_func2_loss_record.append(float(loss2))
 
+        q1_recent_variance = np.var(list(self.q1_record)[-100:])
+        q2_recent_variance = np.var(list(self.q2_record)[-100:])
```
Comment (watakandai) [inline pfrl/agents/goal_conditioned_td3.py]: same here

File: pfrl/agents/goal_conditioned_td3.py [hunk 3]
```diff
@@ -100,8 +107,15 @@ def __init__(
         self.buffer_freq = buffer_freq
         self.minibatch_size = minibatch_size
         self.add_entropy = add_entropy
+
         if add_entropy:
             self.temperature = 1.0
+        self.q_func1_variance_record = collections.deque(maxlen=10)
+        self.q_func2_variance_record = collections.deque(maxlen=10)
```
Comment (watakandai) [inline pfrl/agents/goal_conditioned_td3.py]: why 10 of them? cuz of 10 steps?

File: pfrl/agents/goal_conditioned_td3.py [hunk 4]
```diff
@@ -11,6 +13,11 @@
 from pfrl.utils import clip_l2_grad_norm_
 
 
+def _mean_or_nan(xs):
+    """Return its mean a non-empty sequence, numpy.nan for a empty one."""
+    return np.mean(xs) if xs else np.nan
```
Comment (watakandai) [inline pfrl/agents/goal_conditioned_td3.py]: isnt' this defined somewhere? if not, is there any other good place to put this?

**Review by watakandai — CHANGES_REQUESTED** [review]

File: pfrl/agents/goal_conditioned_td3.py (hunk 3 above)
Comment (peasant98) [reply pfrl/agents/goal_conditioned_td3.py]: just an arbitary value -- I'll make it in argument

Comment (peasant98) [reply pfrl/agents/goal_conditioned_td3.py]: fixed

File: pfrl/agents/hiro_high_level_goal_conditioned_td3.py (hunk 1 above)
Comment (peasant98) [reply pfrl/agents/hiro_high_level_goal_conditioned_td3.py]: fixed

File: pfrl/agents/goal_conditioned_td3.py (hunk 4 above)
Comment (peasant98) [reply pfrl/agents/goal_conditioned_td3.py]: fixed.

**Review by watakandai — APPROVED** [review]

**Comment by watakandai** [comment]
lgtm

**Pull request merged**